docs/fast_data/configuration/projection_storer.md

```diff
@@ -0,0 +1,1048 @@
+in the type expected by the data model for that specific field on which the cast function is applied.
+
+Below is provided an example of cast functions implementation, one for each supported programming language.
+
+<details><summary>Custom Cast Function (Kotlin - mapToAddressType.kts)</summary>
+<p>
+
+```kotlin
+package castFunctions
+
+val addressMapping = mapOf(
+  1 to "SHIPPING",
+  2 to "BILLING",
+  3 to "LIVING",
+)
+
+// NOTE: the name of the function must correspond to
+//       the key associated to the file containing it
+fun mapToAddressType(value: Any, fieldName: String): String? {
+  return when (value) {
+    is String -> addressMapping[value.toInt()]
+    is Int, is Long -> addressMapping[value]
+    else -> {
+      // NOTE: a basic logger can be accesses via internal binding
+      logger.debug("not an address type code: $value - fieldName: $fieldName")
+      
+      null
+    }
+  }
+}
+```
+
+</p>
+</details>
+
+<details><summary>Custom Cast Function (Javascript - castToTitleCase.js)</summary>
+<p>
+
+```javascript
+'use strict'
+
+// NOTE: the name of the function must correspond to
+//       the key associated to the file containing it
+function castToTitleCase(value, fieldName) {
+  const str = value.toString()
+  
+  str[0].toUpperCase() + str.slice(1).toLowerCase()
+}
+```
+
+</p>
+</details>
+
+:::danger
+For those migrating custom cast functions from the ones employed in the Real-Time Updater, please bear in mind that
+the `logger` parameter is not provided anymore in the function signature, and therefore it should be removed.  
+For debugging purposes, a logger is still provided via internal bindings, but it is recommended to not log
+within cast functions at levels higher than `debug`.
+:::
+
+### Consumer
+
+| Property        | Type     | Required | Default |
+|-----------------|----------|----------|---------|
+| `type`          | `string` | &check;  | `kafka` |
+| `configuration` | `object` | &check;  |         |
+
+Describe which type of consumer and its configuration properties the service should employ to read ingestion messages
+as input events. Currently only Kafka (and platforms adopting Kafka APIs) is supported as consumer.
+
+#### Kafka Configuration
+
+When Kafka is selected as consumer for the Projection Storer service, it is possible to provide most of the Kafka Consumer
+properties that are defined in the [Apache Kafka documentation](https://kafka.apache.org/documentation/#consumerconfigs).
+
+This is an example of consumer configuration when `kafka` is selected as type: 
+
+```json
+"consumer": {
+  "type": "kafka",
+  "configuration": {
+    "client.id": "galaxy.fast-data.DEV.inventory-projection-storer-consumer",
+    "bootstrap.servers": "localhost:9092",
+    "group.id": "galaxy.fast-data.DEV.inventory-projection-storer",
+    "auto.offset.reset": "latest",
+    "max.poll.records": 2000,
+    "max.poll.timeout.ms": 500
+  }
+}
+```
+
+:::info
```

Comment:
Personal opinion: I would minimize the usage of the alerts when it is not strictly necessary, having too many alert boxes in the doc diminishes the importance of the more serious stuff

Reply:
I understand, although I believe that infobox help focus the attention on details for keen readers.

This box is to clarify that a specific property is not something they can find in Apache Kafka documentation. It is at info level so that someone can read it as an additional detail and in the box, so that it is sufficiently distinguishable from the main content.